test/primary_failure_and_recovery_test.go

```diff
@@ -205,7 +201,8 @@ var _ = Describe("Primary instances is not available, checking recovery works",
 			// First failover
 			test.whenPrimaryIsDeleted()
 
-			test.thenPodsStatesShouldBe(1, 3)
+			test.thenPodsStatesShouldBe(1, 2+2) // There should be 2 replicas left + wait for at least 2 new replicas to be created because the old replicas don't have replication slots
+			test.thenPodsStatesShouldBe(1, 3)   // after new replicas are created, wait for old replicas to be deleted
```

Comment:
The old behavior is the correct one.
We should expect the final state here, regardless of the intermediate one.
Consider incrementing the validation timeout if required.